sw/device/tests/sim_dv/csrng_fuse_en_sw_app_read.c

```diff
@@ -0,0 +1,225 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_csrng.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/csrng_testutils.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+#include "otp_ctrl_regs.h"                            // Generated
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * OTP HW partition relative IFETCH offset in bytes.
+ *
+ * x = OTP_CTRL_PARAM_EN_SRAM_IFETCH_OFFSET (1728)
+ * y = OTP_CTRL_PARAM_HW_CFG_OFFSET (1664)
+ * IFETCH_OFFSET = (x - y) = 64
+ */
+static const uint32_t kOtpIfetchHwRelativeOffset =
+    OTP_CTRL_PARAM_EN_SRAM_IFETCH_OFFSET - OTP_CTRL_PARAM_HW_CFG_OFFSET;
+
+/**
+ * OTP can only be accessed by 32b aligned addresses. As `csrng_sw_app_read` is
+ * not aligned, we read from the previous aligned address and use the following
+ * offset in bits to access the `csrng_sw_app_read` byte.
+ */
+static const uint32_t kOtpCsrngFwReadBitOffset =
+    (OTP_CTRL_PARAM_EN_CSRNG_SW_APP_READ_OFFSET -
+     OTP_CTRL_PARAM_EN_SRAM_IFETCH_OFFSET) *
+    8;
+
+enum {
+  kExpectedOutputLen = 16,
```

Comment:
Why is this an enum instead of const int?

Reply:
It's broadly used on the opentitan and I'm not sure the main reason. I guess it has better type safity features.

Reply:
@alphan can provide more details here.

Should we update the style guide to capture the rationale?

Reply:
Sure! In a nutshell, We use `enum`s to define _named_ constants that
* appear in the symbol table for easier debugging and don't evaporate like macros do, and
* don't use any memory.

If the constant is defined merely to publish _some_ value, we don't name the type. If, on the other hand, the `enum` has some significance for the API, we name the type (using `typedef`).

`int` is not really an option because: it cannot be used as `case` labels in `switch` statements, and it cannot be used as the dimension of (global) arrays. `int` could be inlined, but depends on the compiler and optimization settings. Arguably, the only advantage of an `int` is you can take its address -- but you can always define one and initialize with an `enum` if needed.

One exception is, if a constant will be used in `asm`, then `#define` is the only option.

See this [SO answer](https://stackoverflow.com/questions/1674032/static-const-vs-define-vs-enum) for more details.

P.S. If you want to try this, please make sure to pass `-Wall -Wextra -Werror -pedantic-errors`.

Reply:
Thanks @alphan, this was very enlightening for me!

Reply:
glad that I could help!